app/apphandlers/embeddedAppHandler.go

```diff
@@ -172,7 +182,7 @@ func (embedded *EmbeddedApp) StartRudderCore(ctx context.Context, options *app.O
 		modeProvider = state.NewStaticProvider(servermode.DegradedMode)
 	}
 
-	proc := processor.New(ctx, &options.ClearDB, gwDBForProcessor, routerDB, batchRouterDB, errDB, multitenantStats, reportingI)
+	proc := processor.New(ctx, &options.ClearDB, gwDBForProcessor, routerDB, batchRouterDB, errDB, multitenantStats, reportingI, transientSources)
```

Comment:
`processor` is the one that initialises `stash`, therefore we need to pass `transientSources` to it.